docs/html/man/man3/GameControls.3
.TH "GameControls" 3 "Version 1.0.0" "Stateless" \" -*- nroff -*-
.ad l
.nh
.SH NAME
GameControls
.SH SYNOPSIS
.br
.PP
.PP
Inherits IInputActionCollection2, and IDisposable\&.
.SS "Classes"

.in +1c
.ti -1c
.RI "struct \fBGamePlayActions\fP"
.br
.ti -1c
.RI "interface \fBIGamePlayActions\fP"
.br
.in -1c
.SS "Public Member Functions"

.in +1c
.ti -1c
.RI "\fBGameControls\fP ()"
.br
.ti -1c
.RI "void \fBDispose\fP ()"
.br
.ti -1c
.RI "bool \fBContains\fP (InputAction action)"
.br
.ti -1c
.RI "IEnumerator< InputAction > \fBGetEnumerator\fP ()"
.br
.ti -1c
.RI "void \fBEnable\fP ()"
.br
.ti -1c
.RI "void \fBDisable\fP ()"
.br
.ti -1c
.RI "InputAction \fBFindAction\fP (string actionNameOrId, bool throwIfNotFound=false)"
.br
.ti -1c
.RI "int \fBFindBinding\fP (InputBinding \fBbindingMask\fP, out InputAction action)"
.br
.in -1c
.SS "Properties"

.in +1c
.ti -1c
.RI "InputActionAsset \fBasset\fP\fC [get]\fP"
.br
.ti -1c
.RI "InputBinding? \fBbindingMask\fP\fC [get, set]\fP"
.br
.ti -1c
.RI "ReadOnlyArray< InputDevice >? \fBdevices\fP\fC [get, set]\fP"
.br
.ti -1c
.RI "ReadOnlyArray< InputControlScheme > \fBcontrolSchemes\fP\fC [get]\fP"
.br
.ti -1c
.RI "IEnumerable< InputBinding > \fBbindings\fP\fC [get]\fP"
.br
.ti -1c
.RI "\fBGamePlayActions\fP \fBGamePlay\fP\fC [get]\fP"
.br
.ti -1c
.RI "InputControlScheme \fBKeyboardScheme\fP\fC [get]\fP"
.br
.ti -1c
.RI "InputControlScheme \fBGamepadScheme\fP\fC [get]\fP"
.br
.in -1c
.SH "Detailed Description"
.PP 
Definition at line \fB18\fP of file \fBGameControls\&.cs\fP\&.
.SH "Constructor & Destructor Documentation"
.PP 
.SS "GameControls\&.GameControls ()"

.PP
Definition at line \fB21\fP of file \fBGameControls\&.cs\fP\&.
.SH "Member Function Documentation"
.PP 
.SS "bool GameControls\&.Contains (InputAction action)"

.PP
Definition at line \fB287\fP of file \fBGameControls\&.cs\fP\&.
.SS "void GameControls\&.Disable ()"

.PP
Definition at line \fB307\fP of file \fBGameControls\&.cs\fP\&.
.SS "void GameControls\&.Dispose ()"

.PP
Definition at line \fB268\fP of file \fBGameControls\&.cs\fP\&.
.SS "void GameControls\&.Enable ()"

.PP
Definition at line \fB302\fP of file \fBGameControls\&.cs\fP\&.
.SS "InputAction GameControls\&.FindAction (string actionNameOrId, bool throwIfNotFound = \fCfalse\fP)"

.PP
Definition at line \fB314\fP of file \fBGameControls\&.cs\fP\&.
.SS "int GameControls\&.FindBinding (InputBinding bindingMask, out InputAction action)"

.PP
Definition at line \fB319\fP of file \fBGameControls\&.cs\fP\&.
.SS "IEnumerator< InputAction > GameControls\&.GetEnumerator ()"

.PP
Definition at line \fB292\fP of file \fBGameControls\&.cs\fP\&.
.SH "Property Documentation"
.PP 
.SS "InputActionAsset GameControls\&.asset\fC [get]\fP"

.PP
Definition at line \fB20\fP of file \fBGameControls\&.cs\fP\&.
.SS "InputBinding? GameControls\&.bindingMask\fC [get]\fP, \fC [set]\fP"

.PP
Definition at line \fB273\fP of file \fBGameControls\&.cs\fP\&.
.SS "IEnumerable<InputBinding> GameControls\&.bindings\fC [get]\fP"

.PP
Definition at line \fB312\fP of file \fBGameControls\&.cs\fP\&.
.SS "ReadOnlyArray<InputControlScheme> GameControls\&.controlSchemes\fC [get]\fP"

.PP
Definition at line \fB285\fP of file \fBGameControls\&.cs\fP\&.
.SS "ReadOnlyArray<InputDevice>? GameControls\&.devices\fC [get]\fP, \fC [set]\fP"

.PP
Definition at line \fB279\fP of file \fBGameControls\&.cs\fP\&.
.SS "InputControlScheme GameControls\&.GamepadScheme\fC [get]\fP"

.PP
Definition at line \fB411\fP of file \fBGameControls\&.cs\fP\&.
.SS "\fBGamePlayActions\fP GameControls\&.GamePlay\fC [get]\fP"

.PP
Definition at line \fB400\fP of file \fBGameControls\&.cs\fP\&.
.SS "InputControlScheme GameControls\&.KeyboardScheme\fC [get]\fP"

.PP
Definition at line \fB402\fP of file \fBGameControls\&.cs\fP\&.

.SH "Author"
.PP 
Generated automatically by Doxygen for Stateless from the source code\&.


Assets/Scripts/Input/GameControls.cs
//------------------------------------------------------------------------------
// <auto-generated>
//     This code was auto-generated by com.unity.inputsystem:InputActionCodeGenerator
//     version 1.6.3
//     from Assets/Scripts/Input/GameControls.inputactions
//
//     Changes to this file may cause incorrect behavior and will be lost if
//     the code is regenerated.
// </auto-generated>
//------------------------------------------------------------------------------

using System;
using System.Collections;
using System.Collections.Generic;
using UnityEngine.InputSystem;
using UnityEngine.InputSystem.Utilities;

public partial class @GameControls: IInputActionCollection2, IDisposable
{
    public InputActionAsset asset { get; }
    public @GameControls()
    {
        asset = InputActionAsset.FromJson(@"{
    ""name"": ""GameControls"",
    ""maps"": [
        {
            ""name"": ""GamePlay"",
            ""id"": ""265c38f5-dd18-4d34-b198-aec58e1627ff"",
            ""actions"": [
                {
                    ""name"": ""Move"",
                    ""type"": ""PassThrough"",
                    ""id"": ""50fd2809-3aa3-4a90-988e-1facf6773553"",
                    ""expectedControlType"": ""Vector2"",
                    ""processors"": """",
                    ""interactions"": """",
                    ""initialStateCheck"": true
                },
                {
                    ""name"": ""look"",
                    ""type"": ""Value"",
                    ""id"": ""c60e0974-d140-4597-a40e-9862193067e9"",
                    ""expectedControlType"": ""Vector2"",
                    ""processors"": """",
                    ""interactions"": """",
                    ""initialStateCheck"": true
                },
                {
                    ""name"": ""Jump"",
                    ""type"": ""Button"",
                    ""id"": ""c55b67a7-8c5d-4be0-a05b-159cfe53f1a1"",
                    ""expectedControlType"": ""Button"",
                    ""processors"": """",
                    ""interactions"": ""Press"",
                    ""initialStateCheck"": false
                },
                {
                    ""name"": ""Fire"",
                    ""type"": ""Button"",
                    ""id"": ""2bfa741f-6359-4df0-90b6-44b6ccc8e09d"",
                    ""expectedControlType"": ""Button"",
                    ""processors"": """",
                    ""interactions"": ""Press"",
                    ""initialStateCheck"": false
                },
                {
                    ""name"": ""Join"",
                    ""type"": ""Button"",
                    ""id"": ""50b4a690-3789-4839-a2cf-545aa630f908"",
                    ""expectedControlType"": ""Button"",
                    ""processors"": """",
                    ""interactions"": """",
                    ""initialStateCheck"": false
                }
            ],
            ""bindings"": [
                {
                    ""name"": """",
                    ""id"": ""e1b8c4dd-7b3a-4db6-a93a-0889b59b1afc"",
                    ""path"": ""<Gamepad>/leftStick"",
                    ""interactions"": """",
                    ""processors"": """",
                    ""groups"": ""Gamepad"",
                    ""action"": ""Move"",
                    ""isComposite"": false,
                    ""isPartOfComposite"": false
                },
                {
                    ""name"": ""WASD"",
                    ""id"": ""0f217c60-3ace-4e8b-b983-f3409c01b486"",
                    ""path"": ""2DVector"",
                    ""interactions"": """",
                    ""processors"": """",
                    ""groups"": """",
                    ""action"": ""Move"",
                    ""isComposite"": true,
                    ""isPartOfComposite"": false
                },
                {
                    ""name"": ""up"",
                    ""id"": ""529ce422-25bd-48c7-989c-68f74185cf72"",
                    ""path"": ""<Keyboard>/w"",
                    ""interactions"": """",
                    ""processors"": """",
                    ""groups"": ""Keyboard"",
                    ""action"": ""Move"",
                    ""isComposite"": false,
                    ""isPartOfComposite"": true
                },
                {
                    ""name"": ""left"",
                    ""id"": ""d5506f6d-f82e-4b93-b401-7ac7c474afc7"",
                    ""path"": ""<Keyboard>/a"",
                    ""interactions"": """",
                    ""processors"": """",
                    ""groups"": ""Keyboard"",
                    ""action"": ""Move"",
                    ""isComposite"": false,
                    ""isPartOfComposite"": true
                },
                {
                    ""name"": ""down"",
                    ""id"": ""9a074fc7-293d-4bb2-b1c4-209a6427bfd2"",
                    ""path"": ""<Keyboard>/s"",
                    ""interactions"": """",
                    ""processors"": """",
                    ""groups"": ""Keyboard"",
                    ""action"": ""Move"",
                    ""isComposite"": false,
                    ""isPartOfComposite"": true
                },
                {
                    ""name"": ""right"",
                    ""id"": ""eba2bd94-450b-4bf1-884b-439b951bdd2d"",
                    ""path"": ""<Keyboard>/d"",
                    ""interactions"": """",
                    ""processors"": """",
                    ""groups"": ""Keyboard"",
                    ""action"": ""Move"",
                    ""isComposite"": false,
                    ""isPartOfComposite"": true
                },
                {
                    ""name"": """",
                    ""id"": ""c106d6e6-2780-47ff-b318-396171bd54cc"",
                    ""path"": ""<Gamepad>/rightStick"",
                    ""interactions"": """",
                    ""processors"": """",
                    ""groups"": ""Gamepad"",
                    ""action"": ""look"",
                    ""isComposite"": false,
                    ""isPartOfComposite"": false
                },
                {
                    ""name"": """",
                    ""id"": ""578caa03-6827-4797-adfc-a59770c437fe"",
                    ""path"": ""<Pointer>/delta"",
                    ""interactions"": """",
                    ""processors"": ""ScaleVector2(x=2,y=2)"",
                    ""groups"": ""Keyboard"",
                    ""action"": ""look"",
                    ""isComposite"": false,
                    ""isPartOfComposite"": false
                },
                {
                    ""name"": """",
                    ""id"": ""49f6af40-3385-4761-8f8a-44b89bd3cd26"",
                    ""path"": ""<Keyboard>/space"",
                    ""interactions"": """",
                    ""processors"": """",
                    ""groups"": ""Keyboard"",
                    ""action"": ""Jump"",
                    ""isComposite"": false,
                    ""isPartOfComposite"": false
                },
                {
                    ""name"": """",
                    ""id"": ""7439f5e2-d30a-451a-9abe-57422a6565da"",
                    ""path"": ""<Gamepad>/buttonNorth"",
                    ""interactions"": """",
                    ""processors"": """",
                    ""groups"": ""Gamepad"",
                    ""action"": ""Jump"",
                    ""isComposite"": false,
                    ""isPartOfComposite"": false
                },
                {
                    ""name"": """",
                    ""id"": ""1d88754b-9faf-46ca-ae21-d21fa295435e"",
                    ""path"": ""<Keyboard>/anyKey"",
                    ""interactions"": """",
                    ""processors"": """",
                    ""groups"": ""Keyboard"",
                    ""action"": ""Join"",
                    ""isComposite"": false,
                    ""isPartOfComposite"": false
                },
                {
                    ""name"": """",
                    ""id"": ""1e76cd86-c94d-44c7-bbf2-28684b35f72e"",
                    ""path"": ""<Gamepad>/start"",
                    ""interactions"": """",
                    ""processors"": """",
                    ""groups"": ""Gamepad"",
                    ""action"": ""Join"",
                    ""isComposite"": false,
                    ""isPartOfComposite"": false
                },
                {
                    ""name"": """",
                    ""id"": ""06fe5bcd-f2af-466e-b2ff-396cc9cf1dfc"",
                    ""path"": ""<Keyboard>/rightCtrl"",
                    ""interactions"": """",
                    ""processors"": """",
                    ""groups"": ""Keyboard"",
                    ""action"": ""Fire"",
                    ""isComposite"": false,
                    ""isPartOfComposite"": false
                },
                {
                    ""name"": """",
                    ""id"": ""58cff9bb-be6e-40ef-82d7-8246ada8ad18"",
                    ""path"": ""<Gamepad>/buttonSouth"",
                    ""interactions"": """",
                    ""processors"": """",
                    ""groups"": ""Gamepad"",
                    ""action"": ""Fire"",
                    ""isComposite"": false,
                    ""isPartOfComposite"": false
                }
            ]
        }
    ],
    ""controlSchemes"": [
        {
            ""name"": ""Keyboard"",
            ""bindingGroup"": ""Keyboard"",
            ""devices"": [
                {
                    ""devicePath"": ""<Keyboard>"",
                    ""isOptional"": false,
                    ""isOR"": false
                }
            ]
        },
        {
            ""name"": ""Gamepad"",
            ""bindingGroup"": ""Gamepad"",
            ""devices"": [
                {
                    ""devicePath"": ""<Gamepad>"",
                    ""isOptional"": true,
                    ""isOR"": false
                }
            ]
        }
    ]
}");
        // GamePlay
        m_GamePlay = asset.FindActionMap("GamePlay", throwIfNotFound: true);
        m_GamePlay_Move = m_GamePlay.FindAction("Move", throwIfNotFound: true);
        m_GamePlay_look = m_GamePlay.FindAction("look", throwIfNotFound: true);
        m_GamePlay_Jump = m_GamePlay.FindAction("Jump", throwIfNotFound: true);
        m_GamePlay_Fire = m_GamePlay.FindAction("Fire", throwIfNotFound: true);
        m_GamePlay_Join = m_GamePlay.FindAction("Join", throwIfNotFound: true);
    }

    public void Dispose()
    {
        UnityEngine.Object.Destroy(asset);
    }

    public InputBinding? bindingMask
    {
        get => asset.bindingMask;
        set => asset.bindingMask = value;
    }

    public ReadOnlyArray<InputDevice>? devices
    {
        get => asset.devices;
        set => asset.devices = value;
    }

    public ReadOnlyArray<InputControlScheme> controlSchemes => asset.controlSchemes;

    public bool Contains(InputAction action)
    {
        return asset.Contains(action);
    }

    public IEnumerator<InputAction> GetEnumerator()
    {
        return asset.GetEnumerator();
    }

    IEnumerator IEnumerable.GetEnumerator()
    {
        return GetEnumerator();
    }

    public void Enable()
    {
        asset.Enable();
    }

    public void Disable()
    {
        asset.Disable();
    }

    public IEnumerable<InputBinding> bindings => asset.bindings;

    public InputAction FindAction(string actionNameOrId, bool throwIfNotFound = false)
    {
        return asset.FindAction(actionNameOrId, throwIfNotFound);
    }

    public int FindBinding(InputBinding bindingMask, out InputAction action)
    {
        return asset.FindBinding(bindingMask, out action);
    }

    // GamePlay
    private readonly InputActionMap m_GamePlay;
    private List<IGamePlayActions> m_GamePlayActionsCallbackInterfaces = new List<IGamePlayActions>();
    private readonly InputAction m_GamePlay_Move;
    private readonly InputAction m_GamePlay_look;
    private readonly InputAction m_GamePlay_Jump;
    private readonly InputAction m_GamePlay_Fire;
    private readonly InputAction m_GamePlay_Join;
    public struct GamePlayActions
    {
        private @GameControls m_Wrapper;
        public GamePlayActions(@GameControls wrapper) { m_Wrapper = wrapper; }
        public InputAction @Move => m_Wrapper.m_GamePlay_Move;
        public InputAction @look => m_Wrapper.m_GamePlay_look;
        public InputAction @Jump => m_Wrapper.m_GamePlay_Jump;
        public InputAction @Fire => m_Wrapper.m_GamePlay_Fire;
        public InputAction @Join => m_Wrapper.m_GamePlay_Join;
        public InputActionMap Get() { return m_Wrapper.m_GamePlay; }
        public void Enable() { Get().Enable(); }
        public void Disable() { Get().Disable(); }
        public bool enabled => Get().enabled;
        public static implicit operator InputActionMap(GamePlayActions set) { return set.Get(); }
        public void AddCallbacks(IGamePlayActions instance)
        {
            if (instance == null || m_Wrapper.m_GamePlayActionsCallbackInterfaces.Contains(instance)) return;
            m_Wrapper.m_GamePlayActionsCallbackInterfaces.Add(instance);
            @Move.started += instance.OnMove;
            @Move.performed += instance.OnMove;
            @Move.canceled += instance.OnMove;
            @look.started += instance.OnLook;
            @look.performed += instance.OnLook;
            @look.canceled += instance.OnLook;
            @Jump.started += instance.OnJump;
            @Jump.performed += instance.OnJump;
            @Jump.canceled += instance.OnJump;
            @Fire.started += instance.OnFire;
            @Fire.performed += instance.OnFire;
            @Fire.canceled += instance.OnFire;
            @Join.started += instance.OnJoin;
            @Join.performed += instance.OnJoin;
            @Join.canceled += instance.OnJoin;
        }

        private void UnregisterCallbacks(IGamePlayActions instance)
        {
            @Move.started -= instance.OnMove;
            @Move.performed -= instance.OnMove;
            @Move.canceled -= instance.OnMove;
            @look.started -= instance.OnLook;
            @look.performed -= instance.OnLook;
            @look.canceled -= instance.OnLook;
            @Jump.started -= instance.OnJump;
            @Jump.performed -= instance.OnJump;
            @Jump.canceled -= instance.OnJump;
            @Fire.started -= instance.OnFire;
            @Fire.performed -= instance.OnFire;
            @Fire.canceled -= instance.OnFire;
            @Join.started -= instance.OnJoin;
            @Join.performed -= instance.OnJoin;
            @Join.canceled -= instance.OnJoin;
        }

        public void RemoveCallbacks(IGamePlayActions instance)
        {
            if (m_Wrapper.m_GamePlayActionsCallbackInterfaces.Remove(instance))
                UnregisterCallbacks(instance);
        }

        public void SetCallbacks(IGamePlayActions instance)
        {
            foreach (var item in m_Wrapper.m_GamePlayActionsCallbackInterfaces)
                UnregisterCallbacks(item);
            m_Wrapper.m_GamePlayActionsCallbackInterfaces.Clear();
            AddCallbacks(instance);
        }
    }
    public GamePlayActions @GamePlay => new GamePlayActions(this);
    private int m_KeyboardSchemeIndex = -1;
    public InputControlScheme KeyboardScheme
    {
        get
        {
            if (m_KeyboardSchemeIndex == -1) m_KeyboardSchemeIndex = asset.FindControlSchemeIndex("Keyboard");
            return asset.controlSchemes[m_KeyboardSchemeIndex];
        }
    }
    private int m_GamepadSchemeIndex = -1;
    public InputControlScheme GamepadScheme
    {
        get
        {
            if (m_GamepadSchemeIndex == -1) m_GamepadSchemeIndex = asset.FindControlSchemeIndex("Gamepad");
            return asset.controlSchemes[m_GamepadSchemeIndex];
        }
    }
    public interface IGamePlayActions
    {
        void OnMove(InputAction.CallbackContext context);
        void OnLook(InputAction.CallbackContext context);
        void OnJump(InputAction.CallbackContext context);
        void OnFire(InputAction.CallbackContext context);
        void OnJoin(InputAction.CallbackContext context);
    }
}
